operator/pkg/model/model.go

```diff
@@ -91,3 +93,11 @@ type BackendPort struct {
 	// matching spec.ports[].name in the target Service.
 	Name string `json:"name,omitempty"`
 }
+
+// GetPort return the string representation of the port (either the port number or the port name)
+func (be *BackendPort) GetPort() string {
+	if be.Port != 0 {
+		return strconv.Itoa(int(be.Port))
+	}
+	return be.Name
```

Comment:
Incidentally, are these mutually exclusive? If this layer assumes the port number takes priority, we should probably make sure that they're either not both configurable or that all other layers prioritize the resolution in the same way.

Reply:
yup, unlike other resources, Ingress resource only allows either port number or port name. k8s API server will reject if both fields are specified.

```
The Ingress "basic-ingress" is invalid: spec.defaultBackend: Invalid value: "": cannot set both port name & port number
```